### ISSUE TITLE
Product grid blocks: 'Sale!' badge should be read by screen readers as 'Product on Sale'

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#1686]
![imatge](https://user-images.githubusercontent.com/3616980/73450539-e15dcc00-4365-11ea-95a6-81b111558d55.png)

**To Reproduce**
1. Create a post with a _Handpicked products_ block.
2. Preview the post in the frontend and navigate it with a screen reader and the keyboard.
3. Notice when you read the `Sale!` badge, it just reads `Sale!`. Ideally it should be `Product on Sale` to make it clear the sale refers to that product.

We implemented the same for the _All Products_ block in: #1656.
